server/index.js

```diff
@@ -71,6 +65,108 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+// GET ALL PRODUCTS IN SHOPPING CART
+app.get('/api/cart', (req, res, next) => {
+
+  if (!req.session.cartId) {
+    return [];
+  } else {
+    const sql = `
+      SELECT
+              "c"."cartItemId",
+              "c"."price",
+              "p"."productId",
+              "p"."image",
+              "p"."name",
+              "p"."shortDescription"
+        FROM  "cartItems" AS "c"
+        JOIN  "products" AS "p" USING ("productId")
+       WHERE  "c"."cartId" = $1
+      `;
+    const params = [req.session.cartId];
+    db.query(sql, params)
+      .then(result => res.json(result.rows))
+      .catch(err => next(err));
+  }
+});
+
+// ADD NEW ITEM INTO SHOPPING CART
+app.post('/api/cart', (req, res, next) => {
+  const productId = req.body.productId;
+  if (!productId) {
+    next(new ClientError('ProductId is a required field', 400));
+  } else if (productId <= 0) {
+    next(new ClientError('ProductId must be an integer greater than 0', 400));
+  }
+
+  const sql = `
+  SELECT  "productId", "price"
+    FROM  "products"
+   WHERE  "productId" = $1
+  `;
+
+  const params = [productId];
+
+  db.query(sql, params)
+    .then(result => {
+      const product = result.rows[0];
+      if (!product) {
+        next(new ClientError(`Cannot find product with ID ${productId}`, 400));
```

Comment:
Here we need to `throw new ClientError(message, statusCode)` to shortcut to our `.catch()` and prevent the rest of the promise chain from attempting to run.